enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+For each kernel module one would provide one CR with the needed information. 
+
+#### Example CR for a hardware vendor (all settings) #2
+
+```
+apiVersion: sro.openshift.io/v1alpha1
+kind: SpecialResource
+metadata:
+  name: <vendor>-<hardware>
+  namespace: <vendor>-<hardware>
+spec:
+  driverContainer:
+    metadata:
+      name: <vendor>-<driver>-<container>
+      namespace: <vendor>-<driver>-<custome>
+    source: 
+      git: 
+        ref: "master"
+        uri: "https://gitlab.com/<vendor>/driver.git"
+    buildArgs:
+      - name: "DRIVER_VERSION"
+        value: "440.64.00"
+      - name: "USE_SPECIFIC_DRIVER_FEATURE"
+        value: "True"
+    artifacts:
+      hostPaths:
+      - sourcePath: "/run/<vendor>/usr/src/<artifact>"
+        destinationDir: "/usr/src/"
+      images:
+      - name: "<vendor>-{{.KernelVersion}}:latest"
+      kind: ImageStreamTag
+      namespace: "<vendor>-<hardware>"
+      pullSecret: "vendor-secret"
+      paths:
+      - sourcePath: "/usr/src/<vendor>/<artifact>
+        destinationDir: "/usr/src/"
+      claims:
+      - name: "<vendor>-pvc"
+      mountPath: "/usr/src/<vendor>-<internal>"
+  node: 
+    selector: "feature.../pci-<VENDOR_ID>.present"
+  dependsOn:
+    - name: <CR_NAME_VENDOR_ID_XYZ>
+    - name: <CR_NAME_VENDOR_ID_KJI>
+```
+
+With the above information XYZ is capable of deducing all needed information to build and manage a DriverContainer. All manifests in XYZ are templates that are templatized during reconciliation with runtime and meta information. 
+
+The name e.g. is used to prefix all resources (Pod, DameonSet, RBAC, ServiceAccount, Namespace, etc) created for this very specific **{vendor}-{hardware}**. The DriverContainer section expects at least the git repository (mandatory) from a vendor. This repository has all tools and scripts to build the kernel module. The base image for a DriverContainer is an UBI7,8  with the KVC (kmod-via-containers)  framework installed. 
+
+KVC provides hooks to build, load, unload the kernel modules and a wrapper for userspace utilities. We might extend the number of hooks to have a similar interface as dkms. 
+Another important field is the build arguments. We have often seen incompatibility between workloads and driver versions. Selecting a specific version is sometimes the only way to have a workload successfully running on OpenShift or BareMetal.  This field can also be used by an administrator to upgrade or downgrade a kernel module due to CVEs, bug fixes or incompatibility. Some drivers have also some flags  to enable or disable specific features of a driver. 
+
+The next section is used to filter the nodes on which a kernel module or driver should be deployed on. It makes no sense to deploy drivers on nodes where the hardware is not available. Furthermore this can also be used to target even subsets of special nodes either by creating labels manually or leveraging NFDs hook functionality. 
```

Comment:
Let's be specific about the "sections." Specifications should be specific to remove ambiguity.